### PR TITLE
Qute - fix default template extensions for java.util.Map

### LINKED ISSUE #6519: Qute - fix default template extensions for java.util.Map
`Map.get()` and `Map.containsKey()` are not handled correctly. A workaround exists:

```java
@TemplateExtension
static Object get(Map<?, ?> map, Object key) {
   return map.get(key);
 }
 @TemplateExtension
 static boolean containsKey(Map<?, ?> map, Object key) {
    return map.containsKey(key);
 }
```

### PR DESCRIPTION
**Pull request by mkouba**
- resolves #6519

### CODE REVIEW

File: extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/BuiltinExtensions.java [hunk 1]
```diff
@@ -29,11 +30,17 @@ public static Object map(Map map, String name) {
                 return map.isEmpty();
             case "get":
                 return map.get(name);
```
Comment (gsmet) [inline extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/BuiltinExtensions.java]: This one is a bit weird, isn't it? You're doing `map.get("get")` AFAICS?

**Review by gsmet — CHANGES_REQUESTED** [review]
Added a small question.

File: extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/BuiltinExtensions.java (hunk 1 above)
Comment (mkouba) [reply extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/BuiltinExtensions.java]: Good catch! This one should be removed...

**Review by gsmet — APPROVED** [review]

File: extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/BuiltinTemplateExtensions.java [hunk 2]
```diff
@@ -27,13 +28,17 @@ public static Object map(Map map, String name) {
             case "empty":
             case "isEmpty":
                 return map.isEmpty();
-            case "get":
-                return map.get(name);
-            case "containsKey":
-                return map.containsKey(name);
             default:
-                return Result.NOT_FOUND;
+                return map.containsKey(name) ? map.get(name) : Result.NOT_FOUND;
```
Comment (gastaldi) [inline extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/BuiltinTemplateExtensions.java]: You can also use `map.getOrDefault(name, Result.NOT_FOUND);` here

Comment (geoand) [reply extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/BuiltinTemplateExtensions.java]: @mkouba are you going to apply this or not?

Comment (mkouba) [reply extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/BuiltinTemplateExtensions.java]: Sure, why not. I'll update the PR in a minute.

**Review by gastaldi — APPROVED** [review]

**Comment by gsmet** [comment]
@mkouba I rebased it to trigger CI, wasn't able to do it manually.

**Pull request merged**